File: 3. Data Science & Algorithms/Sort Algorithms/Selection Sort/SelectionSort.py
def SelectionSort(array):

    def swap(arr, i ,j):
        if i == j:
            pass
        else:
            temp= arr[i]
            arr[i] = arr[j]
            arr[j] = temp

        return arr

    LargestUnsortedIndex= len(array) - 1
    while LargestUnsortedIndex > 0:
        LargestIndex= 0
        for i in range(1, LargestUnsortedIndex+1):
            if array[i] > array[LargestIndex]:
                LargestIndex = i

        swap(array, LargestIndex, LargestUnsortedIndex)

        LargestUnsortedIndex-=1

    return array

print(SelectionSort([35, -22,  15, 1, 7, 20, 55]))

File: 3. Data Science & Algorithms/Sort Algorithms/Selection Sort/SelectionSort.java
package com.company;

public class SelectionSort {

    public static void swap(int[] array, int i, int j)  {
        if (i == j) {
            return;
        }
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    public static void main(String[] args)  {

        int[] intArray = {20, 35, -15, 7,  55, 1, -22};

        for (int LastUnsortedIndex = intArray.length-1; LastUnsortedIndex >0;
                LastUnsortedIndex--)    {

            int LargestIndex = 0;

            for (int i = 1; i <= LastUnsortedIndex; i++)    {
                if (intArray[LargestIndex] < intArray[i])   {
                    LargestIndex = i;

                }

            }

            swap(intArray, LargestIndex, LastUnsortedIndex);

        }


        for (int i = 0; i <intArray.length; i++)    {
            System.out.println(intArray[i]);
        }

    }
}
